src/tagged-type.ts

```diff
@@ -70,16 +73,16 @@ class TaggedTypeEncoded {
     },
     number: (value: number): TaggedDouble | TaggedInt | TaggedLong => {
       if (`${value}`.includes(".")) {
-        return { "@double": value };
+        return { "@double": value.toString() };
```

Comment:
This is a bug:

```
> `${2.1**70}`
'3.5921183035610026e+22'
```

Do you have to transmit it as a string? If so, we need to deliver a string correctly and not in this JS format.

Reply:
we will lose precision here too. We need to make a contract on what is acceptable to us.

example:

```
> 2**100
1.2676506002282294e+30
> 2**100 - 1
1.2676506002282294e+30
>
```

This is what i'm trying to get at with precision loss being treated the same for BigInt and number (e.g. do we allow it or blow up?)